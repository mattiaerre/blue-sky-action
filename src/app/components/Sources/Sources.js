import React from 'react';
import PropTypes from 'prop-types';
import TiNews from 'react-icons/lib/ti/news';
import { NavLink } from 'react-router-dom';
import lablels from '../../../data/labels.json';

const Sources = ({ category, sources }) => (
  <div>
    <div className="row">
      <div className="col-12">
        <h2>
          <TiNews />{' '}
          <span className="align-middle">{lablels.categories[category]}</span>
        </h2>
      </div>
    </div>
    <div className="row pb-3">
      <div className="col-12">
        <ul className="nav nav-pills">
          {sources
            .filter(source => source.category === category)
            .map(source => (
              <li key={source.id} className="nav-item">
                <NavLink
                  className="nav-link"
                  activeClassName="active"
                  to={`/${category}/${source.id}`}
                >
                  {source.name}
                </NavLink>
              </li>
            ))}
        </ul>
      </div>
    </div>
  </div>
);

Sources.propTypes = {
  category: PropTypes.string.isRequired,
  sources: PropTypes.array.isRequired
};

export default Sources;
