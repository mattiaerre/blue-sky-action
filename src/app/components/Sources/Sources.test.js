import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import Sources from './Sources';

describe('<Sources />', () => {
  test('to match snapshot', () => {
    const props = {
      category: 'entertainment',
      sources: [
        {
          category: 'entertainment',
          id: 'buzzfeed',
          name: 'Buzzfeed'
        }
      ]
    };

    const tree = create(
      <StaticRouter context={{}}>
        <Sources {...props} />
      </StaticRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
