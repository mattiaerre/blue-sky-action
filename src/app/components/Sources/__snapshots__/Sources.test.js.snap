// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sources /> to match snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <h2>
        <svg
          baseProfile="tiny"
          className="align-middle"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 4h-18c-1.104 0-2 .896-2 2v12c0 1.104.896 2 2 2h18c1.104 0 2-.896 2-2v-12c0-1.104-.896-2-2-2zm-18 2h8v12h-8v-12zm18 12h-9v-12h9.003l-.003 12zM20 13.5c0-.275-.225-.5-.5-.5h-1c-.275 0-.5.225-.5.5v3c0 .275.225.5.5.5h1c.275 0 .5-.225.5-.5v-3zM17 7.5c0-.275-.225-.5-.5-.5h-3c-.275 0-.5.225-.5.5v5c0 .275.225.5.5.5h3c.275 0 .5-.225.5-.5v-5zM18.5 10h1c.275 0 .5-.225.5-.5s-.225-.5-.5-.5h-1c-.275 0-.5.225-.5.5s.225.5.5.5zM18.5 12h1c.275 0 .5-.225.5-.5s-.225-.5-.5-.5h-1c-.275 0-.5.225-.5.5s.225.5.5.5zM13.5 15h3c.275 0 .5-.225.5-.5s-.225-.5-.5-.5h-3c-.275 0-.5.225-.5.5s.225.5.5.5zM16.5 16h-3c-.275 0-.5.225-.5.5s.225.5.5.5h3c.275 0 .5-.225.5-.5s-.225-.5-.5-.5zM18.5 8h1c.275 0 .5-.225.5-.5s-.225-.5-.5-.5h-1c-.275 0-.5.225-.5.5s.225.5.5.5zM10 7.5c0-.275-.225-.5-.5-.5h-5c-.275 0-.5.225-.5.5v3c0 .275.225.5.5.5h5c.275 0 .5-.225.5-.5v-3zM9.501 14h-5c-.274 0-.5.225-.5.5s.226.5.5.5h5c.274 0 .499-.225.499-.5s-.225-.5-.499-.5zM9.501 12h-5c-.274 0-.5.225-.5.5s.226.5.5.5h5c.274 0 .499-.225.499-.5s-.225-.5-.499-.5zM9.501 16h-5c-.274 0-.5.225-.5.5s.226.5.5.5h5c.274 0 .499-.225.499-.5s-.225-.5-.499-.5z"
          />
        </svg>
         
        <span
          className="align-middle"
        >
          Entertainment
        </span>
      </h2>
    </div>
  </div>
  <div
    className="row pb-3"
  >
    <div
      className="col-12"
    >
      <ul
        className="nav nav-pills"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/entertainment/buzzfeed"
            onClick={[Function]}
          >
            Buzzfeed
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
